core/src/main/java/org/kiwix/kiwixmobile/core/utils/dialog/AlertDialogShower.kt

```diff
@@ -51,6 +66,33 @@ class AlertDialogShower @Inject constructor(private val activity: Activity) :
               ?.invoke()
           }
         }
+        if (url != null) {
+          val textView = TextView(activity.baseContext)
+          textView.setPadding(5, 5, 5, 5)
+          textView.gravity = Gravity.CENTER
+          textView.setLinkTextColor(Color.BLUE)
+          textView.setOnLongClickListener {
+            val clipboard =
+              ContextCompat.getSystemService(activity.baseContext, ClipboardManager::class.java)
+            val clip = ClipData.newPlainText("External Url", "$url")
+            clipboard?.setPrimaryClip(clip)
+            Toast.makeText(
+              activity.baseContext,
+              "External Link Copied to Clipboard",
+              Toast.LENGTH_SHORT
+            ).show()
+            true
+          }
+          textView.text = Html.fromHtml("</br><a href=$url> <b>$url</b>")
```

Comment:
Suggestion: Instead of converting a link to html text, you can always use spannable string, and implement a custom click behaviour here, instead of leaving it to OS.